module/metrics/herocache.go

```diff
@@ -0,0 +1,163 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/onflow/flow-go/module"
+)
+
+const subsystemHeroCache = "hero_cache"
+
+type HeroCacheCollector struct {
+	normalizedBucketSlotAvailableHistogram prometheus.Histogram
+
+	successfulReadCount   prometheus.Counter
+	unsuccessfulReadCount prometheus.Counter
+
+	successfulWriteCount   prometheus.Counter
+	unsuccessfulWriteCount prometheus.Counter
+
+	fullCapacityEntityEjectionCount prometheus.Counter
+	emergencyKeyEjectionCount       prometheus.Counter
+}
```

Comment:
Similar to above, consider more structured names, e.g.:
```
	CountKeyGetSuccess prometheus.Counter
	CountKeyGetFailure prometheus.Counter

	CountKeyPutSuccess prometheus.Counter
	CountKeyPutFailure prometheus.Counter

	CountKeyEjectionAtFullCapacity prometheus.Counter
	CountKeyEjectionDueToEmergency prometheus.Counter
```

Reply:
https://github.com/onflow/flow-go/pull/2093/commits/ce5f7bf2d1adfddda0c749a6bde2aa922883a70d